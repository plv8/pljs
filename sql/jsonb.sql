CREATE FUNCTION get_key(key text, json_data jsonb) RETURNS jsonb
LANGUAGE pljs IMMUTABLE STRICT
AS $$
  var val = json_data[key];
  var ret = {};
  ret[key] = val;
  return JSON.stringify(ret);
$$;

CREATE TABLE jsonbonly (
    data jsonb
);

COPY jsonbonly (data) FROM stdin;
{"ok": true}
\.

-- Call twice to test the function cache.
SELECT get_key('ok', data) FROM jsonbonly;
SELECT get_key('ok', data) FROM jsonbonly;
