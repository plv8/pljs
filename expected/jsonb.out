CREATE FUNCTION get_key(key text, json_data jsonb) RETURNS jsonb
LANGUAGE pljs IMMUTABLE STRICT
AS $$
  var val = json_data[key];
  var ret = {};
  ret[key] = val;
  return JSON.stringify(ret);
$$;
CREATE TABLE jsonbonly (
    data jsonb
);
COPY jsonbonly (data) FROM stdin;
-- Call twice to test the function cache.
SELECT get_key('ok', data) FROM jsonbonly;
     get_key     
-----------------
 "{\"ok\":true}"
(1 row)

SELECT get_key('ok', data) FROM jsonbonly;
     get_key     
-----------------
 "{\"ok\":true}"
(1 row)

